pkg/app/app.go

```diff
@@ -51,11 +58,33 @@ func NewApp(config *config.AppConfig) (*App, error) {
 }
 
 func (app *App) Run() error {
-	err := app.Gui.RunWithSubprocesses()
-
+	err := waitForTerminalSpace()
+	if err != nil {
+		return err
+	}
+	err = app.Gui.RunWithSubprocesses()
 	return err
 }
 
+func waitForTerminalSpace() error {
+	// before we do anything, we need to check that we have some window space available
+	width, height, err := terminal.GetSize(int(os.Stdin.Fd()))
+	if err != nil {
+		return err
+	}
+	if width > 0 && height > 0 {
+		return nil
+	}
+	winch := make(chan os.Signal, 1)
+	signal.Notify(winch, syscall.SIGWINCH)
+	select {
+	case <-winch:
+		return nil
+	case <-time.After(time.Second):
+		return fmt.Errorf("there is no available terminal space")
```

Comment:
Should this string be localization-friendly? Should the first letter be in upper case?

Reply:
I should be removed soon anyway as gocui is getting fixed I think.

Reply:
the gocui fix has been merged into master now @qdm12 so feel free to delete it from this branch :)